From b7f1769a2fe1aa13e3ca5fb4694bb9ac8ea0b93b Mon Sep 17 00:00:00 2001
From: Subrata Chatterjee <schatterjee@habana.ai>
Date: Tue, 04 May 2021 06:07:15 +0300
Subject: [PATCH] [SW-39285] gaudi: security: gic privilege en bit

Set CPU_BOOT_DEV_STS0_GIC_PRIVILEGED_EN bit in status
SP register. LKD on host would read this bit to decide if
GIC interrupt generation from host is blocked.

Change-Id: I7d8329b45b4247f92522b78898c49beffaec6b7c
Signed-off-by: Subrata Chatterjee <schatterjee@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index ab3d34c..301bafc 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1611,6 +1611,7 @@
 	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_SECURITY_EN, is_sec_en);
 
 	cpu_boot_err_clrset(CPU_BOOT_ERR0_SECURITY_NOT_RDY, false);
+	cpu_boot_dev_sts_clrset(CPU_BOOT_DEV_STS0_GIC_PRIVILEGED_EN, true);
 	preboot_puts("Security init finished.\n");
 }
 
